### PR TITLE
Feature/catalog

### PR DESCRIPTION
**Pull request by MarekHalaczkiewicz**

### CODE REVIEW

File: content/pages/catalog.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "title": "Catalog",
+  "description": "Find all the available resources in the Ocean Network."
```
Comment (moritzkirstein) [inline content/pages/catalog.json]: I would suggest changing the description:

`Browse and discover Assets and their Self-Descriptions in the Gaia-X Test Network. Powered by Ocean Protocol.`

File: src/components/pages/Catalog/Tags.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { ReactComponent as Caret } from '../../../images/caret.svg'
+import TagsList from '../../molecules/TagsList'
+import styles from './Tags.module.css'
+import {
+  addExistingParamsToUrl,
+  FilterByTypeOptions
+} from '../../templates/Search/utils'
+import { navigate } from 'gatsby'
+
+const tagsList = [
```
Comment (moritzkirstein) [inline src/components/pages/Catalog/Tags.tsx]: We will need to decide on tags to display here @fr-3deric @oceanByte 
Also in Gaia-X context: should we relabel tags to something like "spaces" "industries" etc. and use related tags in metadata

File: src/components/pages/Catalog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    )
+    return navigate(newUrl)
+  }
+
+  return (
+    <Permission eventType="browse">
+      <>
+        <div className={styles.search}>
+          <div className={styles.row}>
+            <ServiceFilter
```
Comment (moritzkirstein) [inline src/components/pages/Catalog/index.tsx]: I would propose wrapping ServiceFilter and Tags by a div to group the search buttons together on the left side.

**Review by moritzkirstein — COMMENTED** [review]
Love how fast you were able to proceed with the pull request here. I just reviewed most changes and have some general feedback, as well as some more detailed comments within the code.

Let's try to reuse as many components as possible and reduce duplication, where it doesn't create too much overhead (since this still is hackathon related and focus is on speed here)

- I noticed that `src/components/pages/Catalog/filterService.tsx` is a duplicate of `src/components/templates/Search/filterService.tsx` > please remove this
- Same would go for 
    - `src/components/pages/Catalog/filterService.module.css`
    - `src/components/pages/Catalog/sort.tsx`
    - `src/components/pages/Catalog/sort.module.css`
- It would be great if you could also refactor `src/components/organisms/AssetListCatalog.tsx` to use the regular AssetList component (maybe pass a variable table?: boolean if set to true rendering CatalogTable instead of AssetTeasers)
- Additionally, sicne we moved away from publish, history and search pages I think it would be fine if we just use the `src/components/templates/Search/index.tsx` search page template in `src/pages/index.tsx` and merge the changes from `src/components/pages/Catalog/index.tsx` into there (again reducing duplication)

I think we should display more results per page for a browsable catalogue. Please adjust the number per page to something like 20 here.

File: app.config.js [hunk 4]
```diff
@@ -60,5 +60,8 @@ module.exports = {
 
   // Used to show or hide advanced settings button in asset details page
   allowAdvancedSettings: process.env.GATSBY_ALLOW_ADVANCED_SETTINGS || 'false',
-  credentialType: process.env.GATSBY_CREDENTIAL_TYPE || 'address'
+  credentialType: process.env.GATSBY_CREDENTIAL_TYPE || 'address',
+
+  // Address to open assets from main page
+  assetAddress: 'https://portal.minimal-gaia-x.eu'
```
Comment (oceanByte) [inline app.config.js]: suggestion(non-blocking): allow to set an environment variable here as well. The URL might change.

File: content/pages/catalog.json (hunk 1 above)
Comment (oceanByte) [reply content/pages/catalog.json]: suggestion(non-blocking): I like the description by @moritzkirstein. I suggest to replace "Assets" with "Data" though as we are currently discussing the wording with DWF and we should try to get drop all terms that are associated with finance.

File: src/components/molecules/CatalogTable.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  assetsWithPrices
+}: {
+  assetsWithPrices: AssetListPrices[]
+}): ReactElement {
+  return (
+    <Table
+      columns={columns}
+      data={assetsWithPrices}
+      paginationPerPage={assetsWithPrices.length}
+      emptyMessage="The assets will appear here."
```
Comment (oceanByte) [inline src/components/molecules/CatalogTable.tsx]: nitpick, question(non-blocking): this message is unclear. When will the assets appear? Do you want to say `No assets available` or `Assets are loading`?  In addition, we might want to change `asset` with `data` here. Need to be discussed in our daily.

File: src/components/molecules/TagItem.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    if (!checked) {
+      setSelectedTags([...removeItemFromArray(selectedTags, name)])
+    } else {
+      setSelectedTags([...selectedTags, name])
+    }
+  }
+
+  return (
+    <div className={styles.radioWrap} /* key={chainId} */>
+      <label className={styles.radioLabel} /* htmlFor={`opt-${chainId}`} */>
```
Comment (oceanByte) [inline src/components/molecules/TagItem.tsx]: chore(blocking): remove comments or explain why there are still there.

File: src/components/molecules/TagItem.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        <input
+          className={styles.input}
+          // id={`opt-${chainId}`}
```
Comment (oceanByte) [inline src/components/molecules/TagItem.tsx]: chore(blocking): remove comments or explain why there are still there.

File: src/components/molecules/TagItem.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+          type="checkbox"
+          name={name}
+          // value={chainId}
```
Comment (oceanByte) [inline src/components/molecules/TagItem.tsx]: chore(blocking): remove comments or explain why there are still there.

File: src/components/molecules/TagItem.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+          onChange={handleTagSelection}
+          defaultChecked={selectedTags.includes(name)}
+        />
+        {/* <NetworkName key={chainId} networkId={chainId} /> */}
```
Comment (oceanByte) [inline src/components/molecules/TagItem.tsx]: chore(blocking): remove comments or explain why there are still there.

File: src/components/molecules/TagsList.tsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      key={index}
+      name={e.display}
+      selectedTags={selectedTags}
+      setSelectedTags={setSelectedTags}
+    />
+  ))
+
+  return (
+    <>
+      <h4 className={styles.titleGroup}>{title}</h4>
```
Comment (oceanByte) [inline src/components/molecules/TagsList.tsx]: question(non-blocking): are we sure that we want to use h4 here? This has semantic meaning. If you want bold text I'd suggest styling a `<span>`.

File: src/components/organisms/AssetListCatalog.tsx [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  const styleClasses = cx({
+    assetList: true,
+    [className]: className
+  })
+
+  return assetsWithPrices &&
+    !loading &&
+    (isLoading === undefined || isLoading === false) ? (
```
Comment (oceanByte) [inline src/components/organisms/AssetListCatalog.tsx]: nitpick(non-blocking): you could also just check for `!isLoading`. Are you expecting a case where isLoading is `""` or `null`?

File: src/components/pages/Catalog/Tags.module.css [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+  width: var(--font-size-small);
+  height: var(--font-size-small);
+  fill: var(--border-color);
+  margin-left: calc(var(--spacer) / 12);
+  margin-right: -5px;
+  transition: transform 0.2s ease-out;
+  display: none;
+}
+
+@media screen and (min-width: 40rem) {
```
Comment (oceanByte) [inline src/components/pages/Catalog/Tags.module.css]: nitpick(non-blocking): extract 40rem if it is used more often.

File: src/components/pages/Catalog/Tags.module.css [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+  .caret,
+  svg.caret {
+    display: inline-block;
+  }
+}
+
+@media screen and (max-width: 40rem) {
```
Comment (oceanByte) [inline src/components/pages/Catalog/Tags.module.css]: nitpick(non-blocking): extract 40rem if it is used more often.

File: src/components/pages/Catalog/filterService.tsx [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+  FilterByTypeOptions
+} from '../../templates/Search/utils'
+import Button from '../../atoms/Button'
+
+const cx = classNames.bind(styles)
+
+const clearFilters = [{ display: 'Clear', value: '' }]
+
+const serviceFilterItems = [
+  { display: 'data sets', value: FilterByTypeOptions.Data },
```
Comment (oceanByte) [inline src/components/pages/Catalog/filterService.tsx]: nitpick(blocking): I believe you want to display `datasets` and not `data sets`

File: src/components/pages/Catalog/filterService.tsx [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+        await applyServiceFilter(value)
+        setServiceSelections([value])
+      }
+    }
+  }
+
+  async function applyClearFilter() {
+    let urlLocation = await addExistingParamsToUrl(location, ['serviceType'])
+
+    urlLocation = `${urlLocation}`
```
Comment (oceanByte) [inline src/components/pages/Catalog/filterService.tsx]: chore(blocking): Use the `String` function to cast into a string. So you can also use a `const` then.

`const urlLocation = String(await addExistingParamsToUrl(location, ['serviceType']))`

File: src/components/pages/Catalog/index.module.css [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+  width: 100%;
+  white-space: nowrap;
+  margin-bottom: calc(var(--spacer) / 2);
+}
+
+.row > div {
+  margin-bottom: calc(var(--spacer) / 2);
+}
+
+@media (min-width: 40rem) {
```
Comment (oceanByte) [inline src/components/pages/Catalog/index.module.css]: chore(blocking): I see you are using 40rem quite often. This feels like a magic number that is repeating. A variable might be reasonable here.

File: src/components/pages/Catalog/sort.module.css [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+  padding: 0 calc(var(--spacer) / 10);
+  display: flex;
+  align-items: center;
+  border-radius: var(--border-radius);
+  border: 1px solid var(--border-color);
+  background: var(--background-content);
+  overflow-y: auto;
+}
+
+@media screen and (min-width: 40rem) {
```
Comment (oceanByte) [inline src/components/pages/Catalog/sort.module.css]: chore(blocking): extract to variable. See above comments.

File: src/components/pages/Catalog/sort.module.css [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+  .sortList {
+    align-self: flex-end;
+    overflow-y: unset;
+  }
+}
+
+@media screen and (max-width: 40rem) {
```
Comment (oceanByte) [inline src/components/pages/Catalog/sort.module.css]: chore(blocking): extract to variable. See above comments.

File: src/components/pages/Catalog/sort.tsx [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+    if (sortBy) {
+      urlLocation = await addExistingParamsToUrl(location, ['sort'])
+      urlLocation = `${urlLocation}&sort=${sortBy}`
+      setSortType(sortBy)
+    } else if (direction) {
+      urlLocation = await addExistingParamsToUrl(location, ['sortOrder'])
+      urlLocation = `${urlLocation}&sortOrder=${direction}`
+      setSortDirection(direction)
+    }
+    navigate(urlLocation)
```
Comment (oceanByte) [inline src/components/pages/Catalog/sort.tsx]: chore(non-blocking): Refactoring might clean up the code here. Let's spend some minutes after the hackathon to discuss readability and code duplication here.

File: src/components/pages/Catalog/Tags.tsx (hunk 2 above)
Comment (oceanByte) [reply src/components/pages/Catalog/Tags.tsx]: We need some additional support from Kai here. Let's discuss this in our daily.

**Review by oceanByte — CHANGES_REQUESTED** [review]

**Comment by kaimeinke** [comment]
The Gaia-X Federated Service Requirements specify how to filter for data space, see requirements for federated catalogue:

SRS_GXFS_FC_CCF.pdf, page 32, URL: https://www.gxfs.de/federation-services/overview-specification-documents/

Note that, in large federations, it may be necessary to distinguish Self-Descriptions by their application domain or the data space / data ecosystem they originated from, e.g., “healthcare” vs. “logistics”, and to retrieve only Self-Descriptions from one of these domains. This, however, does not require an additional Administrative Metadata field, but is supported because the Core Ontology, which provides the foundation for any Self-Description Schema, provides the gax:theme property, a specialization of dcat:theme from DCAT, which indicates one or more categories that a resource belongs to. Any such category is identified by a URI, which should point to a concept in a Controlled Vocabulary. This may be an external, existing Controlled Vocabulary, or one that is maintained locally using the Schema Storage Subsystem of the Federated Catalogue (cf. Section 4.2).

Furthermore, this leads to the different metadata standards in different spaces later on.

For validating Self-Descriptions, the Federated Catalogue needs to know the schema to validate against. The primary schema is the Gaia-X Self-Description Schema. Supporting the Self-Description Core Ontology (cf. Appendix A) is mandatory for the Federated Catalogue. Further standard Gaia-X Self-Description Schemas Specific ecosystems, specific data spaces, etc., may require additional attributes beyond those defined in the Gaia-X Self-Description Schema. Thus, the Federated Catalogue shall be able to store additional Self-Description Schemas.

Such additional schemas must not define terms in the Gaia-X namespace http://w3id.org/gaia-x/core#. This namespace is reserved for the Gaia-X AISBL. Serializations, e.g., in JSON-LD, must not bind the gax prefix to any namespace other than http://w3id.org/gaia-x/core#.

**Pull request merged**